.github/workflows/sonar.yml

```
@@ -0,0 +1,28 @@
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.24'
+
+      - name: Generate Sonar Report
+        run: go test -coverpkg=./... -coverprofile=coverage.out -json ./... > sonar-report.json
+
+      - name: Upload coverage reports to Sonar
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        if: github.event.pull_request.head.repo.full_name == github.repository || env.SONAR_TOKEN != ''
+        uses: SonarSource/sonarqube-scan-action@v5.1.0
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)